contracts/src/system/Fees.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+interface IFees {
+    function messageFee(uint256 messageSize) external view returns (uint256);
+}
+
+// Contract that will contain fees for contracts that need to apply them
+contract Fees is Initializable, OwnableUpgradeable {
```

Comment:
needs more comments.
If I understand correctly from the PR this is only applied to L2->L1 messages.
And it charges based on the length of the message